tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4940,44 +4941,105 @@ Status ConvertUnpack(OpConverterParams* params) {
   return ConvertSplitHelper(params, inputs.at(0), tf_axis, num, true);
 }
 
-// Supports cast fp16=>fp32 through IIdentityLayer.
+#if IS_TRT_VERSION_GE(7, 0, 0, 11)
+// Supports cast through IIdentityLayer.
 Status ConvertCast(OpConverterParams* params) {
   const NodeDef& node_def = params->node_def;
   TF_RETURN_IF_ERROR(CheckInputsWeights(*params, {{"x", false}}));
-  auto unsupport_cast_error = [&]() {
+
+  auto unsupported_cast_error = [&](DataType in_dtype, DataType out_dtype,
+                                    TrtPrecisionMode precision_mode) {
+    string precision_mode_str;
+    TF_RETURN_IF_ERROR(
+      TrtPrecisionModeToName(precision_mode, &precision_mode_str));
+
     return errors::Unimplemented("Cast op: ", node_def.op(),
-                                 " not supported at: ", node_def.name());
+                                 " not supported at: ", node_def.name(),
+                                 ". Input DType: ", DataTypeString(in_dtype),
+                                 ". Output DType: ", DataTypeString(out_dtype),
+                                 ". Precision Mode: ", precision_mode_str);
   };
 
   DataType input_type;
   TF_RETURN_IF_ERROR(GetInputTfType(*params, &input_type, 0));
-  if (input_type != DataType::DT_HALF) {
-    return unsupport_cast_error();
-  }
 
   DataType output_type;
   TF_RETURN_IF_ERROR(GetNodeDefTfType(params->node_def, &output_type,
                                       kCastOutputTypeAttrName));
 
-  if (output_type != DataType::DT_FLOAT) {
-    return unsupport_cast_error();
+  TrtPrecisionMode precision_mode = params->precision_mode;
+
+  if (isIntTfType(input_type) || isIntTfType(output_type)) {
+    return unsupported_cast_error(
+      input_type,
+      output_type,
+      precision_mode
+    );
+  }
+
+  if (
+    (precision_mode == TrtPrecisionMode::FP32 ||
+    precision_mode == TrtPrecisionMode::NOT_AVAILABLE) &&
+    (input_type != DataType::DT_FLOAT || output_type != DataType::DT_FLOAT)
+  ) {
+        return unsupported_cast_error(
+          input_type,
+          output_type,
+          precision_mode
+        );
   }
 
+  // TODO: @DekhtiarJonathan remove when TRT has fixed the issue
```

Comment:
Can you also add an NVIDIA bug number here?

Reply:
Sure ;)

Reply:
@DEKHTIARJonathan  Any update on this PR? Please. Thanks!